source/extensions/filters/http/aws_lambda/sts_fetcher.cc

```diff
@@ -90,13 +89,17 @@ class StsFetcherImpl : public StsFetcher,
     // Chained assumption specifics 
     const std::string body = fmt::format(StsChainedFormatString, role_arn, now);
     message->body().add(body);
-    aws_authenticator_.init(&creds->accessKeyId().value(), 
+    // this call resets the sha so that our authenticator is in a fresh state
+    // to be reused.
+    auto aws_authenticator = AwsAuthenticator(api_.timeSource(),
+                                             &AWSStsHeaderNames::get().Service);
```

Comment:
You can use regular style here:
```suggestion
    AwsAuthenticator aws_authenticator(api_.timeSource(),
                                             &AWSStsHeaderNames::get().Service);
```